batch/test/test_copy_files.py

```diff
@@ -0,0 +1,547 @@
+import os
+import unittest
+import tempfile
+import subprocess as sp
+import uuid
+import random
+import string
+from shlex import quote as shq
+import google.oauth2.service_account
+from hailtop.auth import get_userinfo
+from hailtop.utils.os import _glob
+
+from batch.google_storage import GCS
+
+key_file = '/gsa-key/key.json'
+project = os.environ['PROJECT']
+user = get_userinfo()
+tmp_bucket = f'gs://{user["bucket_name"]}/test_copy_files'
+
+credentials = google.oauth2.service_account.Credentials.from_service_account_file(key_file)
+gcs_client = GCS(None, project=project, credentials=credentials)
+
+
+class RemoteTemporaryDirectory:
+    def __init__(self):
+        token = uuid.uuid4().hex[:6]
+        self.name = tmp_bucket + f'/{token}'
+
+    def __enter__(self):
+        return self.name
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        remove_remote_dir(self.name)
+
+
+def upload_files(src, dest):
+    os.system(f'gsutil -m -q cp -r {src} {dest}')
+
+
+def move_files(src, dest):
+    os.system(f'gsutil -m -q mv -r {src} {dest}')
+
+
+def remove_remote_dir(path):
+    os.system(f'gsutil -m -q rm -r {path}')
```

Comment:
All of these paths should be shell quoted if we're executing them in a shell.

Actually, we should just use one of the subprocess functions to avoid shell escaping entirely.